### PR TITLE
Update card entity mapping

### PR DESCRIPTION
**Pull request by tomazpu**

### CODE REVIEW

File: src/Entity/Card.php [hunk 1]
```diff
@@ -127,14 +127,14 @@ private function parseFromXml($simpleXml)
         if (isset($simpleXml->{'card-token'}->{'token-id'})) {
             $this->token = $simpleXml->{'card-token'}->{'token-id'};
         }
-        if (isset($simpleXml->{'card-token'}->{'card-type'})) {
-            $this->type = $simpleXml->{'card-token'}->{'card-type'};
+        if (isset($simpleXml->{'card'}->{'card-type'})) {
+            $this->type = $simpleXml->{'card'}->{'card-type'};
```
Comment (flavius) [inline src/Entity/Card.php]: This sets the card type and related to SimpleXML objects.

In M2 Notify.php I get:

`Wirecard\ElasticEngine\Gateway\Model\Notify::saveCreditCardToken {"cardType":"[object] (SimpleXMLElement: visa)"}`

Should be string instead.

Either the getter reaches into the XML and gives back just the string (advantage: keeping xml attributes around), or we discard any additional information attached to the xml and store just the type & co from the beginning.

**Comment by tomazpu** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Coverage increased per file
===========================
+ src/Helper/Enum.php  2
+ src/Response/XmlResponseData.php  1
+ src/BackendService.php  15
+ src/Transaction/Transaction.php  1
+ src/Entity/TransactionDetails.php  1
+ src/Entity/RiskInfo.php  2
+ src/Entity/CustomFieldCollection.php  1
+ src/Entity/CustomField.php  2
         

Coverage decreased per file
===========================
- src/Entity/Browser.php  -2
- src/Entity/Redirect.php  -8
- src/Config/Config.php  -2
- src/TransactionService.php  -2
- src/Entity/Basket.php  -2
- src/Transaction/EpsTransaction.php  -10
- src/Transaction/CustomFieldTransaction.php  -5
- src/Entity/Device.php  -4
- src/Entity/Mappable.php  -1
- src/Entity/Payload/NvpPayloadData.php  -8
- src/Helper/XmlBuilder.php  -10
- src/Response/ResponseFactory.php  -5
- src/Entity/AccountHolder.php  -1
- src/Mapper/RequestMapper.php  -4
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/wirecard/paymentSDK-php/pullRequest?prid=5046666&bid=16618310)

File: src/Entity/Card.php (hunk 1 above)
Comment (tomazpu) [reply src/Entity/Card.php]: i mapped them to strings, can u please check again.

**Comment by tomazpu** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Coverage increased per file
===========================
+ src/Helper/Enum.php  2
+ src/Transaction/Transaction.php  1
+ src/BackendService.php  15
+ src/Response/XmlResponseData.php  1
+ src/Entity/CustomField.php  2
+ src/Entity/TransactionDetails.php  1
+ src/Entity/CustomFieldCollection.php  1
+ src/Entity/RiskInfo.php  2
         

Coverage decreased per file
===========================
- src/Entity/Basket.php  -2
- src/Mapper/RequestMapper.php  -4
- src/Entity/Redirect.php  -8
- src/Helper/XmlBuilder.php  -10
- src/Config/Config.php  -2
- src/Entity/Payload/NvpPayloadData.php  -8
- src/TransactionService.php  -2
- src/Entity/Browser.php  -2
- src/Transaction/CustomFieldTransaction.php  -5
- src/Entity/Device.php  -4
- src/Entity/AccountHolder.php  -1
- src/Response/ResponseFactory.php  -5
- src/Transaction/EpsTransaction.php  -10
- src/Entity/Mappable.php  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/wirecard/paymentSDK-php/pullRequest?prid=5046666&bid=16618310)

**Comment by tomazpu** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Coverage increased per file
===========================
+ src/Transaction/CustomFieldTransaction.php  10
+ src/BackendService.php  21
+ src/Response/ResponseFactory.php  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/wirecard/paymentSDK-php/pullRequest?prid=5046666&bid=16618310)

**Pull request merged**